build-logic/convention/src/main/kotlin/GravatarAndroidLibraryConventionPlugin.kt

```diff
@@ -0,0 +1,40 @@
+import com.android.build.gradle.LibraryExtension
+import com.gravatar.configureDetekt
+import com.gravatar.configureKotlinAndroid
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.configure
+
+class GravatarAndroidLibraryConventionPlugin : Plugin<Project> {
+    override fun apply(target: Project) {
+        with(target) {
+            apply(plugin = "com.android.library")
+            apply(plugin = "org.jetbrains.kotlin.android")
+            apply(plugin = "org.jlleitschuh.gradle.ktlint")
+            apply(plugin = "io.gitlab.arturbosch.detekt")
+
+            extensions.configure<LibraryExtension> {
+                configureKotlinAndroid(this)
+                defaultConfig.apply {
+                    targetSdk = 34
+                    consumerProguardFiles("consumer-rules.pro")
+                }
+                configureBuildTypes()
+            }
+            configureDetekt()
+        }
+    }
+
+    private fun LibraryExtension.configureBuildTypes() {
+        buildTypes {
+            getByName("release") {
+                isMinifyEnabled = false
+                proguardFiles(
+                    getDefaultProguardFile("proguard-android-optimize.txt"),
+                    "proguard-rules.pro",
+                )
```

Comment:
Are we sure that this SDK needs `proguard-rules.pro`?

Reply:
At this point, the `proguard-rules.pro` are empty.

Reply:
> So... we could remove it, but does this setup negatively impact the project?

Let's leave this as it is - I only suggested this to reduce the configuration, but that's not a big deal 👍